uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/BAO/AbsencePeriod.php

```diff
@@ -104,6 +106,34 @@ private static function validateDates($params) {
   }
 
   /**
+   * Checks if another absence period exists with same title as
+   * the absence period being created/updated and throws an exception if found.
+   *
+   * @param array $params
+   *
+   * @throws \CRM_HRLeaveAndAbsences_Exception_InvalidAbsencePeriodException
+   */
+  private static function validateAbsencePeriodTitle($params) {
+    $title = !empty($params['title']) ? $params['title'] : '';
+    $absencePeriod = new self();
+    $absencePeriod->title = $title;
+    $absencePeriod->find(true);
+
+    if (!$absencePeriod->id) {
+      return;
+    }
+
+    $throwExceptionOnCreate = empty($params['id']);
```

Comment:
`$throwExceptionOnCreate` sounds like it will decide whether to throw an exception or not, but really once we get past the previous `if` statement we will be sure to throw an exception every time for creation of a new entity. How about using `$isCreate` for this, and for the other one `$isSeparateUpdate`. I think this would make the following `if` block more understandable.